### PR TITLE
Add min-tls-version flag

### PR DESCRIPTION
**Pull request by bschimke95**
DISA requires a TLS version of 1.2 as a minimum.

### CODE REVIEW

File: src/k8s/pkg/k8sd/setup/kube_apiserver.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func KubeAPIServer(snap snap.Snap, securePort int, nodeIP net.IP, serviceCIDR st
 		"--service-cluster-ip-range":                 serviceCIDR,
 		"--tls-cert-file":                            filepath.Join(snap.KubernetesPKIDir(), "apiserver.crt"),
 		"--tls-cipher-suites":                        strings.Join(apiserverTLSCipherSuites, ","),
+		"--tls-min-version":                          "VersionTLS12",
```
Comment (HomayoonAlimohammadi) [inline src/k8s/pkg/k8sd/setup/kube_apiserver.go:88]: Maybe we can extract this `VersionTLS12`? 
```go
const TLSVersion12 string = "VersionTLS12"
```
I was also thinking that we can also go with 1.3 just to be "even more secure". WDYT?

**Review by HomayoonAlimohammadi — APPROVED** [review]
Thanks a lot @bschimke95. LGTM, just a minor comment.

File: src/k8s/pkg/k8sd/setup/kube_apiserver.go (hunk 1 above)
Comment (bschimke95) [reply src/k8s/pkg/k8sd/setup/kube_apiserver.go:88]: tbh, I don't think that makes much sense and will just cause another indirection. The VersionString does not change. 
I *could* see something like
```
const minTLSVersion string = VersionTLS12
```
but I'd still say that the indirection is not necessary.

As for the 1.3 version: We need to have 1.2 for the DISA but I don't see a reason to further restrict the version (and with that potential users) without a good reason.

Comment (HomayoonAlimohammadi) [reply src/k8s/pkg/k8sd/setup/kube_apiserver.go:88]: Makes sense :+1:

**Comment by bschimke95** [comment]
SBOM etc fail because of other issues, not related to this PR - merging.

**Pull request merged**